File: 이코테_연습문제/dfs_juice.py
r, c = map(int, input().split())

tray = []
for i in range(r):
    tray.append(list(map(int, input())))

print(tray)

# 0 -> 얼음
# 1 -> 칸막이

# 매 번 상하좌우를 확인해보고 방문여부확인 (1은 방문 불가 노드)
def dfs(x, y):
    if x <= -1 or x >= r or y <= -1 or y >= c:
        return False
    if tray[x][y] == 0:
        # 방문 처리
        tray[x][y] = 1
        # 상, 하, 좌, 우 호출 (재귀)
        dfs(x - 1, y)
        dfs(x + 1, y)
        dfs(x, y - 1)
        dfs(x, y + 1)
        return True
    return False

result = 0
for i in range(r):
    for j in range(c):
        if dfs(i, j) == True:
            result += 1

print(result)

File: 이코테_연습문제/DFSJuice.java
public class DFSJuice {

    static int n = 4;
    static int m = 5;
    static int[][] graph = {{0, 0, 1, 1, 0}, {0, 0, 0, 1, 1}, {1, 1, 1, 1, 1}, {0, 0, 0, 0, 0}};
    public static void main (String[] args) {
        int result = 0;

        for (int i = 0; i < n; i++) {
            for (int j = 0; j < m; j++) {
                if (dfs(i, j) == true) result++;
            }
        }

        System.out.println(result);
    }

    public static boolean dfs (int x, int y) {
        if (x < 0 || x >= n || y < 0 || y >= m) return false;

        if (graph[x][y] == 0) {
            graph[x][y] = 1;
            dfs(x-1, y);
            dfs(x+1, y);
            dfs(x, y-1);
            dfs(x, y+1);
            return true;
        }

        return false;
    }
}
